src/connectors/dexalot/dexalot.ts

```diff
@@ -0,0 +1,422 @@
+import {
+  ClobMarketsRequest,
+  ClobOrderbookRequest,
+  ClobTickerRequest,
+  ClobGetOrderRequest,
+  ClobPostOrderRequest,
+  ClobDeleteOrderRequest,
+  ClobGetOrderResponse,
+  ClobBatchUpdateRequest,
+  CreateOrderParam,
+  ClobDeleteOrderRequestExtract,
+} from '../../clob/clob.requests';
+import {
+  CLOBish,
+  MarketInfo,
+  NetworkSelectionRequest,
+  Orderbook,
+} from '../../services/common-interfaces';
+import { BigNumber, Contract, PopulatedTransaction, utils } from 'ethers';
+import { DexalotCLOBConfig } from './dexalot.clob.config';
+import LRUCache from 'lru-cache';
+import {
+  bigNumberWithDecimalToStr,
+  floatStringWithDecimalToFixed,
+} from '../../services/base';
+import { logger } from '../../services/logger';
+import { Avalanche } from '../../chains/avalanche/avalanche';
+import { EVMTxBroadcaster } from '../../chains/ethereum/evm.broadcaster';
+import {
+  createBook,
+  fromUtf8,
+  OrderSide,
+  OrderType1,
+  OrderType2,
+  parseMarkerInfo,
+  parseOrderInfo,
+} from './dexalot.constants';
+import { BalanceRequest } from '../../network/network.requests';
+import { indexOf } from 'lodash';
+import path from 'path';
+import { rootPath } from '../../paths';
+
+export class DexalotCLOB implements CLOBish {
+  private static _instances: LRUCache<string, DexalotCLOB>;
+  private _chain;
+  private _ready: boolean = false;
+  public parsedMarkets: MarketInfo = [];
+  // private _exchangeContract: Contract;
+  private _portfolioContract: Contract;
+  // private _orderBooksContract: Contract;
+  private _tradePairsContract: Contract;
+  // private _gasStationContract: Contract;
+  private _resources: any;
+  private _conf: DexalotCLOBConfig.NetworkConfig;
+  public abiDecoder: any = require('abi-decoder');
+
+  private constructor(network: string) {
+    this._chain = Avalanche.getInstance(network);
+    this._conf = DexalotCLOBConfig.config;
+    this._resources = require(path.join(
+      rootPath(),
+      'src/connectors/dexalot/dexalot_mainnet.json'
+    ));
+    // this._exchangeContract = this.getContract('ExchangeSub', this._resources);
+    this._portfolioContract = this.getContract('PortfolioSub', this._resources);
+    // this._orderBooksContract = this.getContract('OrderBooks', this._resources);
+    this._tradePairsContract = this.getContract('TradePairs', this._resources);
+    // this._gasStationContract = this.getContract('GasStation', this._resources);
```

Comment:
Please do remove these comments, I don't think they are being used or are useful for future reference

Reply:
done